holopy/core/io/io.py

```diff
@@ -380,6 +380,96 @@ def save_image(filename, im, scaling='auto', depth=8):
         pilimage.fromarray(im).save(filename)
 
 
+def save_images(filenames, ims, scaling='auto', depth=8):
+    """
+    Saves a volume as separate images (think of reconstruction volumes).
+
+    Parameters
+    ----------
+    ims : ndarray or :class:`holopy.image.Image`
+        Images to save, with separate z-coordinates from which they each will
+        be selected.
+    filename : str
+        Filename in which to save image. Has to contain a format placeholder
+        for the z-values. E.g. "testfile%f.png"
+    scaling : 'auto', None, or (Int, Int)
+        How the images should be scaled for saving. Ignored for float output.
+        It defaults to auto, use the full range of the output format. Other
+        options are None, meaning no scaling, or a pair of integers specifying
+        the values which should be set to the maximum and minimum values of the
+        image format.
+    depth : 8, 16 or 'float'
+        What type of image to save. Options other than 8bit may not be
+        supported for many image types. You probably don't want to save 8bit
+        images without some kind of scaling.
+    """
+    # TODO: Here could be a check whether ims has the correct format.
+    ims = display_image(ims, scaling)
+    for im in ims:
+        z = im.z.values.item(0)  # TODO: There must be a nicer way…
+        _save_im(filenames % z, im, scaling, depth)
+
+
+def _save_im(filename, im, scaling='auto', depth=8):
+    """
+    Internal single-image-save-method to be used in save_images. Maybe it can
+    be merged with save_image.
```

Comment:
it looks like the original save_image shares a lot of functionality with this, so I think you're right that it can be merged with save image.

If you are up to the challenge, go ahead! If not, don't worry -- I'll make a separate issue and one of me, @barkls , or @ralex0 will take care of it quickly.

Reply:
`save_image` could be rewritten to:
```python
def save_image(filename, im, scaling='auto', depth=8):
    im = display_image(im, scaling)
    _save_im(filename, im, depth)
```

**But**: There is the line `im = im.values` in `_save_im` which in `save_image` is `im = im.values[0]`. As I'm not familiar with the structure of the image-object, someone would need to abstract this line. I could only introduce a boolean parameter flag, split the method up or pass in the values as additional parameter. Each method doesn't seem that great to me. Maybe there is a simple way to check, if there is the need to access the item 0?

Reply:
Let's leave it for a future refactor of io.py then.

Reply:
The `im.values[0]` call is specifically to return only the first slice of a volume reconstruction. This is exactly the problem you are trying to fix with this PR!

Reply:
Thanks for the clarification @barkls! I tried to make a nice adjustment of the method. Please check it out!